### PR TITLE
Add support for IN list filter to PlanBuilder::tableScan

### PR DESCRIPTION
**Pull request by mbasmanova**
Enhance PlanBuilder::tableScan() to add support for "c IN (1, 2, 3)" and "c IN ('a', 'b', 'c')" subfield filters.

### CODE REVIEW

File: velox/exec/tests/utils/PlanBuilder.cpp [hunk 1]
```diff
@@ -315,6 +315,49 @@ std::unique_ptr<common::Filter> makeNotEqualFilter(
...
+      auto elements = arrayVector->elements()->as<SimpleVector<int64_t>>();
+      std::vector<int64_t> values;
+      for (auto i = 0; i < size; i++) {
+        values.push_back(elements->valueAt(offset + i));
+      }
+      return common::test::in(values);
+    }
+    case TypeKind::INTEGER: {
+      auto elements = arrayVector->elements()->as<SimpleVector<int32_t>>();
+      std::vector<int64_t> values;
```
Comment (majetideepak) [inline velox/exec/tests/utils/PlanBuilder.cpp:340]: nit: `int32_t`
I feel templating the case block via `TypeTraits<Kind>::NativeType;` is cleaner.

**Review by majetideepak — COMMENTED** [review]
@mbasmanova thanks for this change.

File: velox/exec/tests/utils/PlanBuilder.cpp (hunk 1 above)
Comment (mbasmanova) [reply velox/exec/tests/utils/PlanBuilder.cpp]: @majetideepak Deepak, I changed this to 

```
  switch (elementType->kind()) {
    case TypeKind::TINYINT:
      return common::test::in(toInt64List<int16_t>(elements, offset, size));
    case TypeKind::SMALLINT:
      return common::test::in(toInt64List<int16_t>(elements, offset, size));
    case TypeKind::INTEGER:
      return common::test::in(toInt64List<int32_t>(elements, offset, size));
    case TypeKind::BIGINT:
      return common::test::in(toInt64List<int64_t>(elements, offset, size));
    case TypeKind::VARCHAR: {
      auto stringElements = elements->as<SimpleVector<StringView>>();
      std::vector<std::string> values;
      for (auto i = 0; i < size; i++) {
        values.push_back(stringElements->valueAt(offset + i).str());
      }
      return common::test::in(values);
    }
    default:
      VELOX_NYI(
          "Unsupported value type for 'in' filter: {}",
          elementType->toString());
  }
```

Would you take another look?

**Review by majetideepak — APPROVED** [review]
@mbasmanova looks nice! Thanks.

**Pull request closed without merging**